### DISCUSSION TITLE
rust-like proc_macro is very important,

### DISCUSSION THREAD

**Discussion by huiseliming** [carbon-language/carbon-lang#1504] [Language design]
New language should implement features that C++ lacks,
Can use macros to visit ast for code generation or replace very useful

**Comment by lexi-nadia** [comment]
Hi! See https://github.com/carbon-language/carbon-lang/pull/89/commits/3b674c4d28873acc28fc89f0024e0cbce1ae1261, which addresses this topic. In short: macros are very powerful but very complex, and if possible, we'd like to have a simpler model for metaprogramming in Carbon.
